src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
+}
```

Comment:
I can clarify in the readme - you either return next or return a response, not both. They're mutually exclusive.

It might be confusing to express folks initially, but there's not a way for me to make it like express. Express is using a callback model (you make a call on the response object and that triggers the production of the response). In this case, we're pretty far removed from that model but intentionally so.

If there are ways to make this difference in thinking clearer in the README, I'm for it.